### PR TITLE
Improve collection breadcrumbs in embedding

### PR DESCRIPTION
**Pull request by WiNloSt**
Epic: #26838
[Product doc](https://www.notion.so/metabase/Improve-collection-breadcrumbs-with-embedding-parameters-51dd704168e344158b649415fb693960)

#### How to test
1. Run Metabase with this branch as an enterprise user (port 3000)
1. Configure enterprise token
1. Enable JWT authentication and make sure you set the secret to `ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff` https://github.com/metabase/sso-examples/blob/master/app-embed-example/server.js#L9-L10
1. Clone https://github.com/metabase/sso-examples
1. Go to `/app-embed-example/`
1. Run `yarn dev` (it will start port 3001 and 3002)
1. Go to http://localhost:3001/analytics and test if embedding shows breadcrumbs as desired.
1. You could try append different parameters to see it in action e.g. http://localhost:3001/analytics?top_nav=false or http://localhost:3001/analytics?breadcrumbs=false or http://localhost:3001/analytics?breadcrumbs=false&search=true
1. Test mobile view.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/26977)

### CODE REVIEW

File: frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx [hunk 1]
```diff
@@ -35,26 +36,31 @@ const AppBarLarge = ({
...
   isSearchVisible,
   isNewButtonVisible,
+  isBreadcrumbsVisible,
   isProfileLinkVisible,
   isCollectionPathVisible,
   isQuestionLineageVisible,
   onToggleNavbar,
   onLogout,
 }: AppBarLargeProps): JSX.Element => {
+  isNavBarOpen = isNavBarOpen && isNavBarVisible;
```
Comment (WiNloSt) [inline frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx:46]: I don't think introducing a new name would be easier to understand here giving that we already have both
- isNavBarOpen (is the sidebar toggled on?)
- isNavBarVisible ([from this bit of code](https://github.com/metabase/metabase/blob/2ff7b99771cf611f33c804b98fb4a5605f454440/frontend/src/metabase/selectors/app.ts#L98-L107))

which are already similar.

And to be fair `isNavBarOpen` (the state alone) isn't that useful. Since if the state is open, but it's not visible then we might as well now show it or say it's not open. That's why I decide to override the parameter here.

File: frontend/src/metabase/nav/components/AppBar/AppBarSmall.styled.tsx [hunk 2]
```diff
@@ -28,6 +28,10 @@ export const AppBarSubheader = styled.div<AppBarSubheaderProps>`
   transition: border-bottom-color 200ms ease;
   border-bottom: 1px solid
     ${props => (props.isNavBarOpen ? color("border") : "transparent")};
+
+  &:empty {
+    display: none;
+  }
```
Comment (WiNloSt) [inline frontend/src/metabase/nav/components/AppBar/AppBarSmall.styled.tsx:34]: Without having to mess around with JS trying to hide the content, this seems more straight forward.

#### With this line
Screenshot 2022-12-05 at 7 33 48 PM


#### Without this line
Screenshot 2022-12-05 at 7 33 57 PM

File: frontend/src/metabase/selectors/app.ts [hunk 3]
```diff
@@ -69,9 +69,13 @@ export const getIsAppBarVisible = createSelector(
     embedOptions,
   ) => {
     const isFullscreen = hash.includes("fullscreen");
+    const allTopNavElementsHidden =
```
Comment (WiNloSt) [inline frontend/src/metabase/selectors/app.ts:72]: [Following this new spec](https://www.notion.so/metabase/Improve-collection-breadcrumbs-with-embedding-parameters-51dd704168e344158b649415fb693960#0ef917ff09244d549fa3a269524c8378), if all top nav elements are hidden then we should hide the top nav too, since it'll serve no purpose.

File: frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx [hunk 4]
```diff
@@ -35,26 +36,31 @@ const AppBarLarge = ({
...
+            onToggleClick={onToggleNavbar}
+          />
+        )}
         <AppBarInfoContainer
           isVisible={!isNavBarOpen || isQuestionLineageVisible}
         >
           {isQuestionLineageVisible ? (
             <QuestionLineage />
-          ) : isCollectionPathVisible ? (
+          ) : isBreadcrumbsVisible && isCollectionPathVisible ? (
```
Comment (ranquild) [inline frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx:63]: WDYT about changing the selector of `isCollectionPathVisible` to take new settings into the account instead of doing here?

File: frontend/src/metabase/nav/components/AppBar/AppBarSmall.styled.tsx (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/nav/components/AppBar/AppBarSmall.styled.tsx:34]: If I get it correctly, by fixing `isCollectionPathVisible` you won't need to do that here

File: frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx [hunk 5]
```diff
@@ -35,26 +36,31 @@ const AppBarLarge = ({
...
+
   return (
     <AppBarRoot isNavBarOpen={isNavBarOpen}>
       <AppBarLeftContainer isNavBarVisible={isNavBarVisible}>
-        <AppBarLogo
-          isNavBarOpen={isNavBarOpen}
-          isToggleVisible={isNavBarVisible}
-          onToggleClick={onToggleNavbar}
-        />
+        {isNavBarVisible && (
```
Comment (ranquild) [inline frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx:51]: Didn't get it why this is changed. This breaks the case with just a logo that is not a button to open the sidebar.

File: frontend/src/metabase/selectors/app.ts [hunk 6]
```diff
@@ -69,9 +69,13 @@ export const getIsAppBarVisible = createSelector(
...
+      !embedOptions.search &&
+      !embedOptions.new_button &&
+      !embedOptions.breadcrumbs;
     if (
       !currentUser ||
-      (isEmbedded && !embedOptions.top_nav) ||
+      (isEmbedded && (!embedOptions.top_nav || allTopNavElementsHidden)) ||
```
Comment (kulyk) [inline frontend/src/metabase/selectors/app.ts:78]: WDYT about extracting this line to a variable or something? IMO it's a bit hard to parse what's going on here

Comment (WiNloSt) [reply frontend/src/metabase/selectors/app.ts:78]: that's fair.

File: frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx (hunk 4 above)
Comment (WiNloSt) [reply frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx:63]: I'll try that, didn't think of that at last night. But seeing this now, it's weird that there are now 2 booleans controlling this component.

File: frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx (hunk 5 above)
Comment (WiNloSt) [reply frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx:51]: I thought we don't want to see the Metabase logo 
image

But agree with you I should revert this back.

File: frontend/src/metabase/nav/components/AppBar/AppBarSmall.styled.tsx (hunk 2 above)
Comment (WiNloSt) [reply frontend/src/metabase/nav/components/AppBar/AppBarSmall.styled.tsx:34]: @ranquild that works. The thing was that there are so many booleans that affect if the breadcrumbs would be showing.

Seems the new approach is to just change how `isCollectionPathVisible` is calculated. And that's that.

File: frontend/src/metabase/redux/embed.js [hunk 7]
```diff
@@ -5,10 +5,11 @@ import {
 } from "metabase/lib/redux";
 
 const DEFAULT_OPTIONS = {
-  top_nav: false,
+  top_nav: true,
```
Comment (WiNloSt) [inline frontend/src/metabase/redux/embed.js:8]: Changing according to [the new spec](https://www.notion.so/metabase/Improve-collection-breadcrumbs-with-embedding-parameters-51dd704168e344158b649415fb693960#0ef917ff09244d549fa3a269524c8378)

File: frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx (hunk 4 above)
Comment (WiNloSt) [reply frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx]: Done!

File: frontend/src/metabase/selectors/app.ts [hunk 8]
```diff
@@ -139,10 +145,17 @@ export const getCollectionId = createSelector(
 );
 
 export const getIsCollectionPathVisible = createSelector(
-  [getQuestion, getDashboard, getRouterPath],
-  (question, dashboard, path) =>
-    ((question != null && question.isSaved()) || dashboard != null) &&
-    PATHS_WITH_COLLECTION_BREADCRUMBS.some(pattern => pattern.test(path)),
+  [getQuestion, getDashboard, getRouterPath, getIsEmbedded, getEmbedOptions],
+  (question, dashboard, path, isEmbedded, embedOptions) => {
+    if (isEmbedded && !embedOptions.breadcrumbs) {
```
Comment (WiNloSt) [inline frontend/src/metabase/selectors/app.ts:150]: Use similar structure as other selectors.
https://github.com/metabase/metabase/blob/b4dfd2c3847c62bb95fdc5aa57ab3b1c9467a465/frontend/src/metabase/selectors/app.ts#L103-L105

File: frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/nav/components/AppBar/AppBarLarge.tsx:44]: I find it confusing, too. WDYT about renaming `isNavBarVisible` to something else, e.g. `isNavBarAvailable` / `isNavBarEnabled` or something like that. Then you could add a variable like `const isNavBarVisible = isNavBarOpen && isNavBarAvailable;` and don't overwrite an existing one

File: frontend/test/metabase/scenarios/embedding/embedding-full-app.cy.spec.js [hunk 9]
```diff
@@ -11,37 +11,49 @@ describe("scenarios > embedding > full app", () => {
...
 
   describe("navigation", () => {
-    it("should hide the top nav by default", () => {
+    it("should show the top nav and breadcrumbs by default", () => {
       visitUrl({ url: "/" });
       cy.findByText("Our analytics").should("be.visible");
+      cy.findByTestId("main-logo").should("be.visible");
+    });
+
+    it("should hide the top nav by a param", () => {
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/embedding/embedding-full-app.cy.spec.js:20]: Add 2 new tests

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/nav/components/AppBar/AppBarLarge.styled.tsx [hunk 10]
```diff
@@ -21,7 +21,7 @@ export const AppBarRoot = styled.div<AppBarRootProps>`
 `;
 
 export interface AppBarLeftContainerProps {
-  isNavBarVisible?: boolean;
+  isNavBarEnabled?: boolean;
```
Comment (WiNloSt) [inline frontend/src/metabase/nav/components/AppBar/AppBarLarge.styled.tsx:24]: Rename the boolean to reflect its behavior.
`isNavBarVisible` means whether or not the navbar is on the screen where as `isNavBarEnabled` means we enabled the navbar but it could be closed.
